pkg/dmaasbackup/periodic_backup_test.go

```diff
@@ -0,0 +1,602 @@
+				Schedule(newTestScheduleSpec()).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:20:00", "", true)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:12:00", v1alpha1.Active, false)).
+				Result(),
+			fakeTime: "2020-06-20 06:21:00",
+			expectedVeleroSchedule: []v1alpha1.VeleroScheduleDetails{
+				veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false),
+				veleroScheduleDetails(t, "2020-06-20 06:12:00", v1alpha1.Deleted, false),
+			},
+		},
+		{
+			name: "should not update old active schedule status on deletion failure",
+			backup: dbkpbuilder().
+				PeriodicConfig("*/8 * * * *", 1).
+				Schedule(newTestScheduleSpec()).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:20:00", "", true)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:12:00", v1alpha1.Active, false)).
+				Result(),
+			fakeTime:            "2020-06-20 06:21:00",
+			scheduleDeleteError: true,
+			shouldErrored:       true,
+			expectedVeleroSchedule: []v1alpha1.VeleroScheduleDetails{
+				veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false),
+				veleroScheduleDetails(t, "2020-06-20 06:12:00", v1alpha1.Active, false),
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				client             = fake.NewSimpleClientset()
+				veleroclient       = velerofake.NewSimpleClientset()
+				veleroFakeInformer = veleroinformer.NewSharedInformerFactory(veleroclient, 0)
+			)
+
+			scheduleInventory := map[string]bool{}
+
+			testTime, err := time.Parse("2006-01-02 15:04:05", test.fakeTime)
+			require.NoError(t, err, "Failed to parse fake time: %v", err)
+
+			veleroclient.PrependReactor("create", "schedules", func(action core.Action) (bool, runtime.Object, error) {
+				obj := action.(core.CreateAction).GetObject().(*velerov1api.Schedule)
+
+				if test.scheduleCreateError {
+					// inject error
+					return true, nil, &apierrors.StatusError{
+						ErrStatus: metav1.Status{Reason: metav1.StatusReasonInvalid},
+					}
+				}
+				// set creationTimestamp
+				setCreationTime := func(obj *velerov1api.Schedule) error {
+					var ntime time.Time
+
+					nslice := strings.Split(obj.Name, "-")
+					if len(nslice) != 2 {
+						return errors.New("invalid name")
+					}
+					ntime, err = time.Parse("20060102150405", nslice[1])
+					if err != nil {
+						return errors.New("invalid name format")
+					}
+					obj.CreationTimestamp.Time = ntime
+					return nil
+				}
+
+				err = setCreationTime(obj)
+				if err != nil {
+					return false, obj, err
+				}
+
+				_, exists := scheduleInventory[obj.Name]
+				if exists {
+					return true, obj, &apierrors.StatusError{
+						ErrStatus: metav1.Status{Reason: metav1.StatusReasonAlreadyExists},
+					}
+				}
+
+				// add schedule to scheduleInventory
+				scheduleInventory[obj.Name] = true
+
+				return true, obj, nil
+			})
+
+			if len(test.existingSchedules) != 0 {
```

Comment:
should this be done first so that error won't happen for creating these existing Schedules?

Reply:
As of now not required. But It may failed if `scheduleCreateError` is set. I've changed the sequence and also added separate reactor for `scheduleCreateError` injection.